frontend/src/utilities/__tests__/useWatchGroups.spec.tsx

```diff
@@ -1,41 +1,156 @@
-import * as React from 'react';
-import { Provider } from 'react-redux';
-import { renderHook } from '~/__tests__/unit/testUtils/hooks';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
 import { GroupsConfig } from '~/pages/groupSettings/groupTypes';
-import { ReduxContext } from '~/redux/context';
-import { store } from '~/redux/store/store';
-import { fetchGroupsSettings } from '~/services/groupSettingsService';
+import { useAppDispatch } from '~/redux/hooks';
+import { fetchGroupsSettings, updateGroupsSettings } from '~/services/groupSettingsService';
 import { useWatchGroups } from '~/utilities/useWatchGroups';
 
 jest.mock('~/services/groupSettingsService', () => ({
   fetchGroupsSettings: jest.fn(),
+  updateGroupsSettings: jest.fn(),
+}));
+// Mock the useDispatch hook
+jest.mock('~/redux/hooks', () => ({
+  useAppDispatch: jest.fn(),
 }));
-
 const fetchGroupSettingsMock = jest.mocked(fetchGroupsSettings);
+const useAppDispatchMock = jest.mocked(useAppDispatch);
+const dispatchMock = jest.fn();
+const updateGroupSettingsMock = jest.mocked(updateGroupsSettings);
+const mockEmptyGroupSettings = {
+  adminGroups: [],
+  allowedGroups: [],
+};
 
 describe('useWatchGroups', () => {
   it('should fetch groups successfully', async () => {
-    const mockEmptyGroupSettings = {
-      adminGroups: [],
+    const mockGroupSettings: GroupsConfig = {
+      adminGroups: [{ id: 1, name: 'odh-admins', enabled: true }],
       allowedGroups: [],
     };
+    updateGroupSettingsMock.mockResolvedValue(mockGroupSettings);
+    useAppDispatchMock.mockReturnValueOnce(dispatchMock);
+    fetchGroupSettingsMock.mockResolvedValue(mockGroupSettings);
+    const renderResult = testHook(useWatchGroups)();
+    expect(renderResult.result.current.updateGroups(mockGroupSettings)).toStrictEqual(undefined);
+    expect(fetchGroupSettingsMock).toHaveBeenCalledTimes(2);
+    expect(renderResult).hookToStrictEqual({
+      groupSettings: mockEmptyGroupSettings,
+      loaded: false,
+      isLoading: true,
+      isGroupSettingsChanged: false,
+      loadError: undefined,
+      updateGroups: expect.anything(),
+      setGroupSettings: expect.anything(),
+      setIsGroupSettingsChanged: expect.anything(),
+    });
+    await renderResult.waitForNextUpdate();
+    expect(fetchGroupSettingsMock).toHaveBeenCalledTimes(2);
+    expect(renderResult).hookToStrictEqual({
+      groupSettings: mockGroupSettings,
+      loaded: false,
+      isLoading: false,
+      isGroupSettingsChanged: false,
+      loadError: new TypeError('dispatch is not a function'),
+      updateGroups: expect.anything(),
+      setGroupSettings: expect.anything(),
```

Comment:
Expect the return result to be a function.
```suggestion
      setGroupSettings: expect.any(Function),
```